### DISCUSSION TITLE
Responsive Views

### DISCUSSION THREAD

**Discussion by chrbox** [GuavaCZ/calendar#39] [Ideas]
It would be nice if there was a way to change the view ‘dayGridMonth’ to ‘listMonth’ on small screens.

**Comment by lukas-frey** [comment]
You could use alpineJS's `x-resize` directive to check for your desired screen size and then call `$wire.setOption('view', '<your desired view>')`.

Might need some tinkering, but it could work.

**Comment by TheChosenOne29** [comment]
@chrbox Hi, did you find a solution for this?

**Comment by drelich** [comment]
In case it's still relevant, I use `mobiledetect/mobiledetectlib` for exactly that function. In my calendar class file, I then do:

```
public function getCalendarView(): string 
    {
        $detect = new MobileDetect();
        return ($detect->isMobile() && !$detect->isTablet()) 
            ? 'timeGridWeek'
            : 'timeGridWeek';
    }
```
    
 And to fine tune the design itself, I simply use CSS media query...
